controllers/ocsinitialization/ocsinitialization_controller.go

```diff
@@ -252,14 +255,82 @@ func (r *OCSInitializationReconciler) ensureRookCephOperatorConfigExists(initial
 			Namespace: initialData.Namespace,
 		},
 	}
-	err := r.Client.Create(r.ctx, rookCephOperatorConfig)
-	if err != nil && !errors.IsAlreadyExists(err) {
-		r.Log.Error(err, fmt.Sprintf("Failed to create %s configmap", util.RookCephOperatorConfigName))
+
+	csiPluginDefaults := defaults.DaemonPlacements["csi-plugin"]
+	csiProvisionerDefaults := defaults.DaemonPlacements["csi-provisioner"]
+
+	pluginTolerations := r.getCsiPluginTolerations()
+	provisionerTolerations := r.getCsiProvisionerTolerations()
+
+	opResult, err := ctrl.CreateOrUpdate(r.ctx, r.Client, rookCephOperatorConfig, func() error {
+
+		if rookCephOperatorConfig.Data == nil {
+			rookCephOperatorConfig.Data = make(map[string]string)
+		}
+
+		if pluginTolerations != nil {
+			tolerations := append(pluginTolerations, csiPluginDefaults.Tolerations...)
+			tolerationsYAML, err := yaml.Marshal(tolerations)
+			if err != nil {
+				return err
+			}
+			rookCephOperatorConfig.Data["CSI_PLUGIN_TOLERATIONS"] = string(tolerationsYAML)
+			initialData.Status.RookCephOperatorConfig.CsiPluginTolerationsModified = true
+		} else if pluginTolerations == nil && initialData.Status.RookCephOperatorConfig.CsiPluginTolerationsModified {
+			delete(rookCephOperatorConfig.Data, "CSI_PLUGIN_TOLERATIONS")
+			initialData.Status.RookCephOperatorConfig.CsiPluginTolerationsModified = false
+		}
+
+		if provisionerTolerations != nil {
+			tolerations := append(provisionerTolerations, csiProvisionerDefaults.Tolerations...)
+			tolerationsYAML, err := yaml.Marshal(tolerations)
+			if err != nil {
+				return err
+			}
+			rookCephOperatorConfig.Data["CSI_PROVISIONER_TOLERATIONS"] = string(tolerationsYAML)
+			initialData.Status.RookCephOperatorConfig.CsiProvisionerTolerationsModified = true
+		} else if provisionerTolerations == nil && initialData.Status.RookCephOperatorConfig.CsiProvisionerTolerationsModified {
+			delete(rookCephOperatorConfig.Data, "CSI_PROVISIONER_TOLERATIONS")
+			initialData.Status.RookCephOperatorConfig.CsiProvisionerTolerationsModified = false
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		r.Log.Error(err, "Failed to create/update rook-ceph-operator-config configmap", "OperationResult", opResult)
```

Comment:
I believe opResult will always be None if the func hit an error as such it can be removed from the log.

Reply:
Let it be, It does not harm

Reply:
sry, it doesn't provide any value either, if we really want to know operationresult, it's better to log after the err check which provides you info whether the resource is created or updated or none.

Reply:
changed it